client/manager_sql.go

```diff
@@ -253,6 +258,13 @@ func (m *SQLManager) CreateSchemas(dbName string) (int, error) {
 func (m *SQLManager) GetConcreteClient(ctx context.Context, id string) (*Client, error) {
 	var d sqlData
 	if err := m.DB.GetContext(ctx, &d, m.DB.Rebind("SELECT * FROM hydra_client WHERE id=?"), id); err != nil {
+		if errorsx.Cause(err) == sql.ErrNoRows {
+			return nil, errors.WithStack(&herodot.DefaultError{
```

Comment:
You can use `herodot.ErrUnauthorized.WithReason("The requested OAuth 2.0 client does not exist or you did not provide the necessary credentials.")`